nilearn/datasets/func.py

```diff
@@ -2076,35 +2083,44 @@ def fetch_development_fmri(n_subjects=None, reduce_confounds=True,
     # Participants data: ids, demographics, etc
     participants = _fetch_development_fmri_participants(data_dir=data_dir,
                                                         url=None,
-                                                        verbose=verbose)
+                                                        verbose=verbose
+                                                        )[::-1] # so adults come first
```

Comment:
Do we need to do this given [L2121](https://github.com/nilearn/nilearn/pull/2035/files#diff-472cc8a2f83316878a7f02c131580611R2121) ?

Reply:
Yeah -- unfortunately the next line does not actually pay attention to the order of the hstack. When I was doing it locally, it wasn't actually changing the order

Reply:
That's odd.... maybe I can dig into that.

Reply:
Looks like the "problem" comes in at `np.in1d`. It returns a sorted array, and since children have the earlier IDs, they're returned first.One idea, then would be:

```python
participants = participants[np.in1d(participants['participant_id'], ids)][ids.argsort()]
```

Since we'd be passing the indices of the unsorted array, this would return the list with the ids in the ordering generated from `np.hstack`. 

WDYT ?

Reply:
I'm into it! I'll make the adjustment

Reply:
hmm.. that didn't work for me actually when I ran it locally.

Reply:
Weird -- it works for me locally. What are you getting back ?

Reply:
Sorry should've been more specific. I did: 
```
ds = datasets.fetch_development_fmri(n_subjects=5)
```
and I got 
```
array([('sub-pixar002',  4.8569473 , '4yo', 'child', 'F', 'R'),
       ('sub-pixar003',  4.15331964, '4yo', 'child', 'F', 'R'),
       ('sub-pixar004',  4.47364819, '4yo', 'child', 'F', 'R'),
       ('sub-pixar123', 27.06      , 'Adult', 'adult', 'F', 'R'),
       ('sub-pixar001',  4.77481177, '4yo', 'child', 'M', 'R')],
      dtype=[('participant_id', '<U12'), ('Age', '<f8'), ('AgeGroup', '<U6'), ('Child_Adult', '<U5'), ('Gender', '<U4'), ('Handedness', '<U4')])
```

Reply:
Hmm. That doesn't match what I'm seeing...

But this is really small so I don't want to hold this PR up over it !

Reply:
🤔 
I'm too jetlagged to figure this out. i'll try again in the morning perhaps?

Reply:
Did this question get resolved? (and are y'all over your jetlag? :smile: )

Reply:
@emdupre and I met and solved it! I think this one is all set @GaelVaroquaux !